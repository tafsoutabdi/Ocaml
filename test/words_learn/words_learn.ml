open Tokenization.Words.Tokenizer
open Utils.Assoc
open Utils.Std

let runtest batch = 
  Format.printf "batch = %s@.learn batch = %s@.@."
    (string_of_list string_of_string batch)  
    (string_of_vocabulary (learn batch))

let () = runtest ["hello world"]

let () = runtest ["Le \"coupe-faim,\""; "c'est pour la pause."; ""]