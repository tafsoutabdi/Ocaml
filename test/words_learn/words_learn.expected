batch = ["hello world"]
learn batch = [("hello", 0); (" ", 1); ("world", 2)]

batch = ["Le "coupe-faim,""; "c'est pour la pause."; ""]
learn batch = [("Le", 0); (" ", 1); (""", 2); ("coupe", 3); ("-", 4); ("faim", 5); (",", 6); ("c", 7); ("\'", 8); ("est", 9); ("pour", 10); ("la", 11); ("pause", 12); (".", 13)]

